enhancements/testing/openshift-tests-extension.md

```diff
@@ -0,0 +1,642 @@
+---
+title: extended-platform-tests
+authors:
+  - "@jupierce"
+  - "@stbenjam"
+reviewers:
+  - "@deads2k"
+creation-date: 2024-09-05
+last-updated: 2024-09-05
+status: implementable
+superseded-by:
+  - "enhancements/testing/openshif-tests-extension.md"
+---
+
+# OpenShift Tests Extensions
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Define a pattern by which repos other than origin can contribute tests to 
+OpenShift test suites.
+// Layered on commits from Stephen with good description from gdoc if we move forward.
+
+## Motivation
+
+Increase the number of tests in the OpenShift Tests Suite by reducing the
+effort, complexity, and risk of introducing new tests.
+
+### Goals
+
+1. Reduce the effort, complexity, and risk of introducing new tests to OpenShift.
+2. Allow tests to be introduced in the same pull request as the PR which fixes a bug or introduces a feature.
+3. Allow component owners to introduce new test suites / add tests to existing suites.
+4. Allow non-OpenShift components, including layered products, to participate in extending OpenShift tests.
+5. Maintain and increase testing coverage.
+6. Maintain and increase code quality.
+7. Maintain and improve the signal TRT provides through Component Readiness.
+8. Provide a standard model for testing component configurations.
+9. Allow distributed contribution of tests while introducing centralized mechanisms to require important tests from component owners and improve coverage over time.
+10. Improve testing efficiency by reducing initialization time.
+11. Allow layered products to participate in OpenShift testing & component readiness.
+
+### Non-Goals
+1. Fully decompose `openshift-tests` or decentralize test orchestration & reporting.
+
+## Proposal
+
+### Concepts
+
+#### Component
+A component is a part of a software product. Component information includes the product name, 
+component name, and the type of the component. A single invocation of an external binary 
+can only act on a single component.
+
+#### Subcomponent
+A further refinement of a component, if necessary. When combined with Component information,
+it should be sufficient to file a Jira ticket that will reach domain experts. 
+
+#### Test ID
+A unique combination of Component, Subcomponent, and test name. A Test ID should not be
+repeated in a test listing across all participating extensions.

```

Comment:
I say that `testHash` is a digest of test IDs -- it is a method of checking for poorly interacting tests. Test ID is just a string that unique identifies a unit of test logic across time. I was thinking this would be `<product>-<component type>-<component>-<original test name>` . 
openshift/origin gets to come up with the actual format of the string they want to use based on the information returned by the tests extension framework implementation. 
The onus of maintaining the original test name as metadata in the extension is placed on the component owner.